### ISSUE TITLE
SignalConst not support LegendText

### ISSUE THREAD

**Issue by fanshuxian** [ScottPlot/ScottPlot#4081]
**Issue:** (Describe the bug here)
I have a `SignalConst` plot (not using `Signal` due to the large amount of data). Since there are multiple curves, I need to display a legend. However, after setting the `LegendText` and executing `ShowLegend`, it does not work as expected.

**ScottPlot Version:** 
5.0.36

code:
```csharp
MainPlot.Plot.Clear();
B_Root.Child = MainPlot;
var hl = MainPlot.Plot.Add.HorizontalLine(0);
hl.IsDraggable = true;
hl.LineWidth = 1;

for (int i = 0; i < OMM.SelectedTask.ChartData.Length; i++)
{
    var sign = MainPlot.Plot.Add.SignalConst(OMM.SelectedTask.ChartData[i].ToArray(), OMM.SelectedTask.XPeriod);
    sign.LegendText = OMM.SelectedTask.ChannelName?[i];
}
MainPlot.Plot.ShowLegend();
MainPlot.Plot.XLabel("Time(ms)", 15);
MainPlot.Plot.YLabel("Voltage(V)", 15);
MainPlot.Plot.Axes.Left.Min = -2;
MainPlot.Plot.Axes.Left.Max = 2;
var buff = OMM.SelectedTask.WaveData.SampleTime * 0.0625;
MainPlot.Plot.Axes.Bottom.Min = 0 - buff;
MainPlot.Plot.Axes.Bottom.Max = OMM.SelectedTask.WaveData.SampleTime + buff;
MainPlot.Plot.Font.Automatic();
MainPlot.Refresh();
```

chart:

![image](https://github.com/user-attachments/assets/c1b067f1-18f3-4cf8-83ea-fbc41b70cdcf)

**Comment by KroMignon** [comment]
@fanshuxian thank you for reporting this!

In fact there is a bug in `SignalConst`, I will submit as PR to solve this. I hope this will be included in next release!

**Comment by swharden** [comment]
Thanks for reporting this @fanshuxian, and for the fix @KroMignon! I merged #4082 and this will indeed be fixed in the next release 🚀
